src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -362,12 +359,26 @@ public void testNoModuleFactory() throws Exception {
     @Test
     public void testFinishLocalSetupFullyInitialized() throws Exception {
         final Checker checker = new Checker();
-        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
+        final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        checker.setModuleClassLoader(contextClassLoader);
         final PackageObjectFactory factory = new PackageObjectFactory(
-            new HashSet<>(), Thread.currentThread().getContextClassLoader());
+            new HashSet<>(), contextClassLoader);
         checker.setModuleFactory(factory);
-
+        checker.setBasedir("testBaseDir");
+        checker.setLocaleLanguage("it");
+        checker.setLocaleCountry("IT");
         checker.finishLocalSetup();
+
+        final Context context = (Context) Whitebox.getInternalState(checker, "childContext");
+        assertEquals(System.getProperty("file.encoding", "UTF-8"), context.get("charset"));
+        assertEquals(contextClassLoader, context.get("classLoader"));
+        assertEquals("error", context.get("severity"));
+        assertEquals("testBaseDir", context.get("basedir"));
+
+        final Field sLocale = LocalizedMessage.class.getDeclaredField("sLocale");
```

Comment:
I am not familar with the utility, so you can ignore this then. I just thought since we used it once, we could do it again.